protocol/message_handler.go

```diff
@@ -1,68 +1,127 @@
 package protocol
 
 import (
+	"fmt"
 	"github.com/pkg/errors"
+	"strings"
+	"time"
 
+	"github.com/status-im/status-go/eth-node/crypto"
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/protocol/protobuf"
 	v1protocol "github.com/status-im/status-go/protocol/v1"
 )
 
-type persistentMessageHandler struct {
-	persistence *sqlitePersistence
+var defaultSystemMessagesTranslations = map[protobuf.MembershipUpdateEvent_EventType]string{
```

Comment:
I would probably just create a function:
```go
type membershipUpdateArgs struct {
    From string
    Name string
    Members []string
    Member string
}

func membershipUpdateMessage(e v1protocol.MembershipUpdateEvent) string {
    switch e.Type {
    case protobuf.MembershipUpdateEvent_CHAT_CREATED:
        return fmt.Sprintf("%s created the group %s", e.From, e.Name)
    ...
    }
    
}
```

Reply:
They need to be passed in as config though, as we need to support multiple languages, it's not pretty, but I think it's useful to generate system-messages in status-go (so that also console client can use them and does not have to re-implement for example), but in status-react we support multiple languages, so that's the best I can come up with.

otherwise we could have passed system messages to status-react, then s.r. would translate and save them, but is more convoluted, as you can't just pass them as messages, you need to have a new type as you need to carry some of the info that a message does not have.

A third option would be to generate system messages in status-react (as before) and save them, but then any client needs to do something similar, and I think it's quite useful, and you have this mix of processing between s.r. and s.go.

Reply:
Kinda understand but don't really have a full picture :) 

Maybe we can extract it to a separate file then? I feel like it can still be improved but the first step is to maybe better encapsulate and separate it from the main logic.

Reply:
yes, I can put it in a `group_chat_system_messages.go`, it should be a bit clearer, thanks

Reply:
separated